Makefile

```diff
@@ -29,6 +29,21 @@ build-docker:
 		--build-arg VERSION \
 		--build-arg BUILD_NUMBER \
 		--target build .
+	@docker tag aws-env docker.sendgrid.net/sendgrid/aws-env
+
+.PHONY: push
+push: 
+	docker push docker.sendgrid.net/$(NAMESPACE)/$(APPNAME)
+
+.PHONY: push-tagged
+push-tagged:
+	docker tag docker.sendgrid.net/$(NAMESPACE)/$(APPNAME) docker.sendgrid.net/$(NAMESPACE)/$(APPNAME):$(VERSION)
+	docker push docker.sendgrid.net/$(NAMESPACE)/$(APPNAME):$(VERSION) 
+
+.PHONY: push-pre-tagged
+push-pre-tagged:
+	docker tag docker.sendgrid.net/$(NAMESPACE)/$(APPNAME) docker.sendgrid.net/$(NAMESPACE)/$(APPNAME):pre${BUILD_NUMBER}
+	docker push docker.sendgrid.net/$(NAMESPACE)/$(APPNAME):pre${BUILD_NUMBER}
```

Comment:
this could be a semver, e.g. `:v0.0.0-alpha+${BUILD_NUMBER}` (see https://semver.org/#spec-item-10). Instead of v0.0.0 it could be the actual pending version if that's readily available.

Reply:
I like that, unfortunately we don't have access to the 'upcoming' version, but we can use the sentinel v0.0.0.